### PR TITLE
HSEARCH-4000 @IndexingDependency, @AssociationInverseSide should be repeatable

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4000

### CODE REVIEW

File: integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/AnnotationInverseSideIndexingDependencyRepatableIT.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.IndexingDependency;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.ObjectPath;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.PropertyValue;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.test.annotation.TestForIssue;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+public class AnnotationInverseSideIndexingDependencyRepatableIT {
```
Comment (yrodiere) [inline integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/AnnotationInverseSideIndexingDependencyRepatableIT.java:33]: There's a typo here.

Anyway, you could rename this to simply `DependencyRepeatableIT`, to mirror `DependencyIT`? Or maybe `DependencyMultiExtractionIT`?

**Review by yrodiere — COMMENTED** [review]
Looks good overall, but I would clarify the javadoc.

**Comment by fax4ever** [comment]
Thanks for the suggestions. I applied them.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**